src/commands/GanacheCommands.ts

```diff
@@ -12,8 +12,32 @@ import {Telemetry} from '../TelemetryClient';
 import {ProjectView} from '../ViewItems';
 
 export namespace GanacheCommands {
-  // Command to bind to UI commands
-  export async function startGanacheCmd(projectView?: ProjectView): Promise<void> {
+  /**
+   * Represents the configuration options to spawn a new Ganache server.
+   */
+  export type GanachePortAndOptions = {
+    /**
+     * The `port` where the Ganache process will be listening to.
+     */
+    port: number;
+
+    /**
+     * Additional `options` such as forking information used by the Ganache process.
+     */
+    options: TLocalProjectOptions;
+  };
+
+  /**
+   * Starts a new Ganache server using the `ganacheConfigSelector`.
+   * The `ganacheConfigSelector` resolves to the `port` and `options`
+   * used to start a new Ganache process.
+   * The selector must be lazy since `startGanacheCmd` needs to first check
+   * whether some required apps are installed.
+   *
+   * @param ganacheConfigSelector the selector that resolves to a `port` and `options`.
+   * @returns
+   */
+  export async function startGanacheCmd(ganacheConfigSelector: () => Promise<GanachePortAndOptions>): Promise<void> {
```

Comment:
Maybe keep the `projectView` as the input parameter like the `stopGanacheCmd(projectView?: ProjectView): Promise<void>` function just to keep the notation, but for me it's perfect :)

Reply:
I also was keen to keep both `startGanacheCmd` and `stopGanacheCmd` with the same signature. However, they get the `projectView` in different ways, since when restarting, the `port` and `options` from `stopGanacheCmd` flows into `startGanacheCmd`.